src/transformers/trainer_tf.py

```diff
@@ -273,45 +284,44 @@ def train(self) -> None:
         logger.info("  Num Epochs = %d", epochs)
         logger.info("  Total optimization steps = %d", self.train_steps)
 
+        self.global_step = 0
+        self.epoch = 0
```

Comment:
Can you elaborate more on why this variable plz? Why not taking the epoch value just above.

Reply:
This is a float value showing number of epochs elapsed.
When comparing training runs of different batch size, or different data size, you will have different number of steps. You can plot loss vs epoch and superpose multiple runs this way.

![image](https://user-images.githubusercontent.com/715491/83760909-3888eb80-a63b-11ea-9e95-57bc1324d1cf.png)

Reply:
OH I see!!! I didn't know that (sorry I'm not really familiar with wandb) it ok makes sense.

Reply:
Would be nice if you can give a more elaborate name instead of just `epoch` (to do not confuse with the loop epoch variable) and add a comment explaining that it is used only for wandb.

Reply:
Ok, I just kept the same name as the one used in `Trainer`.
Any suggestion? `epoch_elapsed` would also bring confusion.
Another possibility would be to change the loop epoch to `epoch_iter`. What do you think?

Reply:
I suggest `epoch_logging` and indeed `epoch_iter` changing both + a small comment for `epoch_logging`